README.md

```diff
@@ -0,0 +1,120 @@
+## `fnord`
+
+This repo contains an in-progress prototype of some of the
+foundational aspects of V2 of Kubernetes Federation.  fnord builds on the
+sync controller (a.k.a. push reconciler) from [Federation
+V1](https://github.com/kubernetes/federation/) to iterate on the API
+concepts laid down in the [brainstorming
+doc](https://docs.google.com/document/d/159cQGlfgXo6O4WxXyWzjZiPoIuiHVl933B43xhmqPEE/edit#)
+and further refined in the [architecture
+doc](https://docs.google.com/document/d/1ihWETo-zE8U_QNuzw5ECxOWX0Df_2BVfO3lC4OesKRQ/edit#).
+Access to both documents is available to members of the
+[kubernetes-sig-multicluster google group](https://groups.google.com/forum/#!forum/kubernetes-sig-multicluster).
+
+<p align="center"><img src="docs/images/propagation.png" width="711"></p>
+
+## Concepts
+
+fnord uses the following abstractions to support the propagation of a
+logical federated type:
+
+- Template: defines the representation of the resource common across clusters
+- Placement: defines which clusters the resource is intended to appear in
+- Override: optionally defines per-cluster field-level variation to apply to the template
+
+These 3 abstractions provide a concise representation of a resource
+intended to appear in multiple clusters.  Since the details encoded by
+the abstractions are the minimum required for propagation, they are
+well-suited to serve as the glue between any given propagation
+mechanism and higher-order behaviors like policy-based placement and
+dynamic scheduling.
+
+## Working with fnord
+
+### Required: `apiserver-builder`
+fnord depends on
+[apiserver-builder](https://github.com/kubernetes-incubator/apiserver-builder)
+to generate code and build binaries.  Download a [recent
+release](https://github.com/kubernetes-incubator/apiserver-builder/releases)
+and install it in your `PATH`.
+
+### Adding a new API type
+
+As per the
+[docs](https://github.com/kubernetes-incubator/apiserver-builder/blob/master/docs/tools_user_guide.md#create-an-api-resource)
+for apiserver-builder, bootstrapping a new fnord API type can be
+accomplished as follows:
+
+```
+# Bootstrap and commit a new type
+$ apiserver-boot create group version resource --group federation.k8s.io --version v1alpha1 --kind <your-kind>
```

Comment:
--group federation~~.k8s.io~~ since `k8s.io` is added and `--group must match regex ^[a-z]+$ but was (federation.k8s.io)`

Reply:
Good catch!